File: IOT/Anshul-13J_AnshulNigam_2024cs1069_2/FunctionAndArrays/AnyBaseAddition.cpp
#include<iostream>

using namespace std;

int getSum(int b, int n1, int n2) {
    int carry=0,i=0,j=0,k;
    int d1[5]={},d2[5]={},sum[5],SUM=0;
    while(n1)
    {
        d1[i++]=n1%10;
        n1/=10;
    }
    while(n2)
    {
        d2[j++]=n2%10;
        n2/=10;
    }
    for(k=0; k<i || k<j; k++)
    {
        sum[k]=(d1[k]+d2[k]+carry)%b  ;
     //   cout<<d1[k]<<" "<<d2[k]<<" "<<carry<<" "<<sum[k]<<endl;
        carry= (d1[k]+d2[k]+carry)/b;
    }
    if(carry)
        sum[k++]=carry;
    
    for(i=k-1;i>=0;i--)
    {
        SUM=SUM*10+ sum[i];
    }
    return SUM;
    
}


int main() {
    int b, n1, n2;
    cin >> b >> n1 >> n2;
    cout << getSum(b, n1, n2) << endl;
}

File: IOT/anshulgurjar53_AnshulSinghGurjar_2024it1193_2/FunctionAndArrays/anybaseaddition.java
import java.util.*;
import java.io.*;
  
  public class Main{
  
  public static void main(String[] args) {
      Scanner scn = new Scanner(System.in);
      int b = scn.nextInt();
      int n1 = scn.nextInt();
      int n2 = scn.nextInt();
  
      int d = getSum(b, n1, n2);
      System.out.println(d);
   }
  
   public static int getSum(int b, int n1, int n2){
       int rem1=0,rem2=0,x=0,c=0,quo=0,carry=0,num=0;
       while(n1!=0||n2!=0||carry!=0)
       {
           rem1=n1%10;
           rem2=n2%10;
           x=rem1+rem2+carry;
           if(x>=b)
           {
               carry=x/b;
               quo=(x%b);
               quo=quo*(int)Math.pow(10,c++);
           }
           else{
               carry=0;
               quo=x*(int)Math.pow(10,c++);
           }
           
           
           num+=quo;
           n1=n1/10;
           n2=n2/10;
       }
       return num;

   }
  }